cmd/scan/scan.go

```
@@ -0,0 +1,220 @@
+				}
+			}
+		}
+	}
+
+	return addr
+}
+
+func ExistTx(addr Address, tx Tx) bool {
+	for _, t := range addr.Txs {
+		if t == tx {
+			return true
+		}
+	}
+	return false
+}
+
+func UpdateAddressInfo(addrs []Address, deps []Deposit, blockID int64) []Address {
+	branch := 0
+	for i, addr := range addrs {
+		branch = 0
+
+		if addr.MaxScanBlock == 0 && branch == 0 && blockID > 1 {
+			addr = CompareAddress(addr, deps)
+			addrs[i].Txs = addr.Txs
+			addrs[i].MaxScanBlock = blockID
+			addrs[i].MidScanBlock = blockID
+			branch = 1
+		}
+
+		if addr.MinScanBlock < addr.MaxScanBlock && addr.MinScanBlock == (blockID-1) && branch == 0 {
+			addr = CompareAddress(addr, deps)
+			addrs[i].Txs = addr.Txs
+			addrs[i].MinScanBlock = blockID
+			if addrs[i].MinScanBlock > addrs[i].MidScanBlock {
+				addrs[i].MidScanBlock = addrs[i].MinScanBlock
+			}
+			branch = 1
+		}
+
+		if addr.MaxScanBlock > addr.MinScanBlock && addr.MaxScanBlock == (blockID-1) && branch == 0 {
+			addr = CompareAddress(addr, deps)
+			addrs[i].Txs = addr.Txs
+			addrs[i].MaxScanBlock = blockID
+			branch = 1
+		}
+
+		if addr.MinScanBlock == addr.MidScanBlock && addr.MinScanBlock == addr.MaxScanBlock && addr.MaxScanBlock == (blockID-1) && branch == 0 {
+			addr = CompareAddress(addr, deps)
+			addrs[i].Txs = addr.Txs
+			addrs[i].MaxScanBlock = blockID
+			addrs[i].MidScanBlock = blockID
+			addrs[i].MinScanBlock = blockID
+			branch = 1
+		}
+
+	}
+	return addrs
+}
+
+func LoadWallet(file string) ([]Address, error) {
+	var addrs []Address
+	wallet, err := os.Open(file)
+	defer wallet.Close()
+	if err != nil {
+		return nil, err
+	}
+	jsonParser := json.NewDecoder(wallet)
+	jsonParser.Decode(&addrs)
+	return addrs, nil
+}
+
+func SaveWallet(file string, addrs []Address) error {
+	wallet, err := os.Open(file)
+	defer wallet.Close()
+	if err != nil {
+		return err
+	}
+	addrsJson, _ := json.MarshalIndent(addrs, "", "    ")
+	err = ioutil.WriteFile(file, addrsJson, 0644)
```

should return the error

---

done